### PR TITLE
Refactored pushStack function

### PR DESCRIPTION
**Pull request by sromexs**
Hi I refactored if else conditions to more readable version.

ts

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| buildDuration | 11.2s | 11.2s | ⚠️ +7ms |
| nodeModulesSize | 42.8 MB | 42.8 MB | -433 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.069 | 2.103 | ⚠️ +0.03 |
| / avg req/sec | 1208.29 | 1189.04 | ⚠️ -19.25 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.527 | 1.416 | -0.11 |
| /error-in-render avg req/sec | 1637.28 | 1765.23 | +127.95 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| 677f882d2ed8..a2e7.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.8 kB | 59.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 622 B | 622 B | ✓ |
| withRouter.html gzip | 608 B | 608 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| buildDuration | 13.5s | 13.2s | -381ms |
| nodeModulesSize | 42.8 MB | 42.8 MB | -433 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| 677f882d2ed8..a2e7.js gzip | 13.4 kB | 13.4 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 6.65 kB | 6.65 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.8 kB | 59.8 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _app-2a09aa2..4a98.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8b758..aef6.js gzip | 3.46 kB | 3.46 kB | ✓ |
| amp-e3e9bc99..932c.js gzip | 536 B | 536 B | ✓ |
| hooks-5023ef..3670.js gzip | 888 B | 888 B | ✓ |
| index-3a2b03..c677.js gzip | 227 B | 227 B | ✓ |
| link-177412b..043e.js gzip | 1.67 kB | 1.67 kB | ✓ |
| routerDirect..1a66.js gzip | 303 B | 303 B | ✓ |
| withRouter-9..1b7d.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.66 kB | 8.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _buildManifest.js gzip | 347 B | 347 B | ✓ |
| Overall change | 347 B | 347 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _error.js | 1.02 MB | 1.02 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| 500.html | 2.65 kB | 2.65 kB | ✓ |
| amp.amp.html | 10.5 kB | 10.5 kB | ✓ |
| amp.html | 1.86 kB | 1.86 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1.02 MB | 1.02 MB | ✓ |
| link.js | 1.08 MB | 1.08 MB | ✓ |
| routerDirect.js | 1.07 MB | 1.07 MB | ✓ |
| withRouter.js | 1.07 MB | 1.07 MB | ✓ |
| Overall change | 5.27 MB | 5.27 MB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 5 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| buildDuration | 13.2s | 13.3s | ⚠️ +70ms |
| nodeModulesSize | 42.8 MB | 42.8 MB | -433 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.998 | 2.005 | ⚠️ +0.01 |
| / avg req/sec | 1251.53 | 1246.93 | ⚠️ -4.6 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.5 | 1.437 | -0.06 |
| /error-in-render avg req/sec | 1666.93 | 1739.53 | +72.6 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| 597-2bc2376a..203d.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 6.59 kB | 6.59 kB | ✓ |
| webpack-HASH.js gzip | 954 B | 954 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _app-aedc815..1421.js gzip | 1.26 kB | 1.26 kB | ✓ |
| _error-71ec2..1a96.js gzip | 3.38 kB | 3.38 kB | ✓ |
| amp-33a09cb0..6745.js gzip | 536 B | 536 B | ✓ |
| hooks-4e968a..f870.js gzip | 902 B | 902 B | ✓ |
| index-5c6845..f75c.js gzip | 230 B | 230 B | ✓ |
| link-99f0c6c..b84a.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..bb56.js gzip | 306 B | 306 B | ✓ |
| withRouter-7..2133.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.57 kB | 8.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| _buildManifest.js gzip | 326 B | 326 B | ✓ |
| Overall change | 326 B | 326 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | sromexs/next.js refactore-pushStack | Change |
| - | - | - | - |
| index.html gzip | 585 B | 585 B | ✓ |
| link.html gzip | 592 B | 592 B | ✓ |
| withRouter.html gzip | 581 B | 581 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/",
         "query": {},
-        "buildId": "7IGHtBMRxtbHuzoe9ER-t",
+        "buildId": "bSmRNpD_UrpTlom_qdAc2",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/7IGHtBMRxtbHuzoe9ER-t/_buildManifest.js"
+      src="/_next/static/bSmRNpD_UrpTlom_qdAc2/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/7IGHtBMRxtbHuzoe9ER-t/_ssgManifest.js"
+      src="/_next/static/bSmRNpD_UrpTlom_qdAc2/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -48,7 +48,7 @@
         "props": { "pageProps": {} },
         "page": "/link",
         "query": {},
-        "buildId": "7IGHtBMRxtbHuzoe9ER-t",
+        "buildId": "bSmRNpD_UrpTlom_qdAc2",
         "isFallback": false,
         "gip": true
       }
@@ -82,11 +82,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/7IGHtBMRxtbHuzoe9ER-t/_buildManifest.js"
+      src="/_next/static/bSmRNpD_UrpTlom_qdAc2/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/7IGHtBMRxtbHuzoe9ER-t/_ssgManifest.js"
+      src="/_next/static/bSmRNpD_UrpTlom_qdAc2/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -43,7 +43,7 @@
         "props": { "pageProps": {} },
         "page": "/withRouter",
         "query": {},
-        "buildId": "7IGHtBMRxtbHuzoe9ER-t",
+        "buildId": "bSmRNpD_UrpTlom_qdAc2",
         "isFallback": false,
         "gip": true
       }
@@ -77,11 +77,11 @@
       async=""
     ></script>
     <script
-      src="/_next/static/7IGHtBMRxtbHuzoe9ER-t/_buildManifest.js"
+      src="/_next/static/bSmRNpD_UrpTlom_qdAc2/_buildManifest.js"
       async=""
     ></script>
     <script
-      src="/_next/static/7IGHtBMRxtbHuzoe9ER-t/_ssgManifest.js"
+      src="/_next/static/bSmRNpD_UrpTlom_qdAc2/_ssgManifest.js"
       async=""
     ></script>
   </body>
```
</details>
</details>
Commit: 2eea19e451fa50cf00b55762ae3cf523a5e04663

File: packages/next/build/tracer.ts [hunk 1]
```diff
@@ -9,20 +9,14 @@ export function stackPush(compiler: any, spanName: string, attrs?: any): any {
...
-        span = tracer.startSpan(spanName, attrs ? attrs() : undefined)
-      })
-    } else {
-      span = tracer.startSpan(spanName, attrs ? attrs() : undefined)
-    }
   }
 
+  !stack && compilerStacks.set(compiler, (stack = []))
+  const parent = stack[stack.length - 1]
+  parent ? tracer.withSpan(parent, fillSpan) : fillSpan()
```
Comment (shuding) [inline packages/next/build/tracer.ts:18]: Let's keep the current if-else structure IMO, that's more readable.

**Comment by sromexs** [comment]
I made some changes i think now that's more readable.

ts

**Review by ijjk — COMMENTED** [review]
Hi, this tracing has been removed so I think we can close this now, thanks for the PR though!

**Pull request closed without merging**